### ISSUE TITLE
ReflectionParameter incorrectly reports allowsNull as true

### ISSUE THREAD

**Issue by calvinbaart** [peachpiecompiler/peachpie#416]
In the following case:
```php
$tmp1 = function (stdClass $user) {
    return true;
};

$tmp2 = function (?stdClass $user) {
    return true;
};

$param1 = (new ReflectionFunction($tmp1))->getParameters()[0];
$param2 = (new ReflectionFunction($tmp2))->getParameters()[0];

echo $param1->allowsNull() ? "true\r\n" : "false\r\n";
echo $param2->allowsNull() ? "true\r\n" : "false\r\n";
```

PHP will output
```
false
true
```

When running the same code in PeachPie the output will be
```
true
true
```

Laravel uses Reflection to determine if it has enough info to run a given callback and fails in the case of PeachPie
